penny_university_frontend/src/selectors/chats.ts

```diff
@@ -0,0 +1,34 @@
+import{ RootState } from '../reducers'
+import * as selectEntities from './entities'
+import Chat from '../models/chat'
+
+const getChatByID = (state: RootState, id: number) => {
+  const chats = selectEntities.getChats(state)
+  const chat = chats[id]
+  if (chat) {
+    // Make sure Participants are always sorted with Organizer first
+    chat.participants.sort((a: Participant, b: Participant) => {
+      if (a.role === 'Organizer') return -1
+      if (b.role === 'Participant') return 1
+      return 0
+    })
+  }
+  return chat || new Chat()
+}
```

Comment:
Yeah, selectors could be cached so that is one advantage. They also encapsulate the logic that you are using to access your store so you don't have to rewrite it in every view. You could also write unit tests for them and it cleans up the logic in the mapStateToProps functions.